### PR TITLE
🚇🩹 Fix CI

### PR DESCRIPTION
**Pull request by s-weigand**
This combines #243 and #244 since both changes are needed for the CI to pass.

### Change summary

- [🚇🩹 Fix GHA permissions](https://github.com/glotaran/pyglotaran-extras/commit/18bb1ceb4dc9155bc654cde87ac2375440ded33e)
- [🚇👌 Improve integration test artifact cleanup order](https://github.com/glotaran/pyglotaran-extras/commit/b944dee8131c021884d463ac51fa7c94609416d5)
- [🧪🩹 Fix deprecation waring test by filtering warning category](https://github.com/glotaran/pyglotaran-extras/commit/1b5391fcb66e31eb8e77ad32ae1dbd0df0565412)
- [🧰 Update sourcery config to new syntax](https://github.com/glotaran/pyglotaran-extras/commit/ff909d86a48fc40c4e89452555f2faa6ba6a5c6f)

### Checklist

- [x] ✔️ Passing the tests (mandatory for all PR's)

### CODE REVIEW

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -68,14 +70,15 @@ jobs:
...
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          name: example-plots-*
-
       - name: Upload Example Plots Artifact
         uses: actions/upload-artifact@v4
         with:
           name: example-plots
           path: example-plots
+          overwrite: true
```
[the remark this answers is not in the document]
Comment (s-weigand) [reply .github/workflows/integration-tests.yml:78]: This is intendet

File: .sourcery.yaml [hunk 2]
```diff
@@ -1,7 +1,9 @@
 ignore: []
 
-refactor:
-  skip: [simplify-boolean-comparison]
+rule_settings:
+  disable:
```
[the remark this answers is not in the document]
Comment (s-weigand) [reply .sourcery.yaml:4]: Explicit > Implicit

**Comment by s-weigand** [comment]
The CI is currently failing because the changed permissions from fork PRs are not applied in the PR itself since that is a security risk.
Here is an [example of the PR passing on my fork](https://github.com/s-weigand/pyglotaran-extras/pull/17)

**Review by jsnel — APPROVED** [review]
Ack. that all workflows are passing after security permissions have been updated.

**Pull request merged**